docs/transform.ipynb

```diff
@@ -116,61 +410,90 @@
+    "*   **cirq.eject_phased_paulis**: Pushes X, Y, and PhasedX gates towards the end of the circuit, potentially absorbing Z gates and modifying gates along the way.\n",
+    "*   **cirq.eject_z**:  Pushes Z gates towards the end of the circuit, potentially adjusting phases of gates that they pass through.\n",
+    "*   **cirq.expand_composite**:  Uses `cirq.decompose` to expand composite gates.\n",
+    "*   **cirq.merge_k_qubit_unitaries**: Replaces connected components of unitary operations, acting on <= k qubits, with op-tree given by `rewriter(circuit_op)`.\n",
+    "*   **cirq.optimize_for_target_gateset**: Attempts to convert a circuit into and equivalent circuit using only gates from a given target gateset.\n",
+    "*   **cirq.stratified_circuit:**: Repacks the circuit to ensure that moments only contain operations from the same category.\n",
+    "*   **cirq.synchronize_terminal_measurements**:  Moves all terminal measurements in a circuit to the final moment, if possible."
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {
-    "id": "c6c7e3ed57ba"
+    "id": "iIHUZvLRlHFj"
    },
    "source": [
-    "### Create your own optimizers\n",
+    "## Compiling to NISQ targets: `cirq.CompilationTargetGateset`\n",
+    "Cirq's philosophy on compiling circuits for execution on a NISQ target device or simulator is that it would often require running only a handful of individual compilation passes on the input circuit, one of the other. \n",
+    "\n",
+    "**`cirq.CompilationTargetGateset`** is an abstraction in Cirq to represent such compilation targets as well as the bundles of transformer passes which should be executed to compile a circuit to this target. Concretely speaking, a compilation target gateset is:\n",
+    "* A subclass of `cirq.Gateset`, which can be used for description and validation of gates supported by a specific target device / simulator.\n",
+    "* A bundle of pre/post transformers + an analytical decomposition method that can be used to rewrite an optimized equivalent circuit using gates from this target gateset. \n",
     "\n",
-    "You can create your own optimizers to transform and modify circuits to fit hardware, gate sets, or other requirements.  Optimizers can also be used to generate noise.  See [noise](noise.ipynb) for details.\n",
+    "Cirq has implementations for common target gatesets like `cirq.CZTargetGateset`, `cirq.SqrtIswapTargetGateset` etc.\n",
     "\n",
-    "You can do this by implementing the function `optimize_circuit`.\n",
     "\n",
-    "If your optimizer is a local optimizer and depends primarily on operator being examined, you can alternatively inherit `cirq.PointOptimizer` and implement the function `optimization_at(self, circuit, index, op)` that optimizes a single operation.\n",
+    "**`cirq.optimize_for_target_gateset`** is a transformer which compiles a given circuit for a `cirq.CompilationTargetGateset` gateset via the following steps:\n",
```

Comment:
It feels like there might be a bit too much information in this text here. Could we cut this down to introduce `optimize_for_target_gateset` maybe demo a built-in gateset and end on some hook to encourage the reader to check out other CompilationTargetGatesets to see how they work ?

Reply:
Reduced the information content and restructured. PTAL.